pkg/vault/contracts/Vault.sol

```diff
@@ -1171,25 +1182,21 @@ contract Vault is IVaultMain, VaultCommon, Proxy {
 
         // If it's a query, the Vault may not have enough underlying tokens to wrap. Since in a query we do not expect
         // the sender to pay for underlying tokens to wrap upfront, return the calculated amount without checking for
-        // the imbalance.
+        // the imbalance. Also, the operation will be reverted, so the balance of the buffers do not matter.
         if (_isQueryContext()) {
-            return (amountInUnderlying, amountOutWrapped);
+            return (amountInUnderlying, amountOutWrapped, 0, 0);
         }
 
         bytes32 bufferBalances = _bufferTokenBalances[wrappedToken];
 
         if (bufferBalances.getBalanceDerived() >= amountOutWrapped) {
             // The buffer has enough liquidity to facilitate the wrap without making an external call.
-            uint256 newDerivedBalance;
+            bufferUnderlyingBalance = bufferBalances.getBalanceRaw() + amountInUnderlying;
             unchecked {
                 // We have verified above that this is safe to do unchecked.
-                newDerivedBalance = bufferBalances.getBalanceDerived() - amountOutWrapped;
+                bufferWrappedBalance = bufferBalances.getBalanceDerived() - amountOutWrapped;
             }
-
-            bufferBalances = PackedTokenBalance.toPackedBalance(
-                bufferBalances.getBalanceRaw() + amountInUnderlying,
-                newDerivedBalance
-            );
+            bufferBalances = PackedTokenBalance.toPackedBalance(bufferUnderlyingBalance, bufferWrappedBalance);
```

Comment:
I mean this looks good in general, but the stakes are too high at this point. 

I don't feel comfortable modifying all of these lines just to modify a few fields in an event.

How about just returning the packed balances and unpacking outside? Maybe not the most efficient, but you won't need any extra (warm) storage reads and the diff will be even smaller.

Reply:
The function now returns 4 arguments, there are multiple if / else levels, early returns, etc. If we stare too much at the function we might break it.